### DISCUSSION TITLE
Mantine Form and Zod type issue

### DISCUSSION THREAD

**Discussion by stj-be** [mantinedev/mantine#5552] [Q&A] [answered]
Im stuck in a bit of a catch 22 here.

I have a `<Select />` component which should be empty but required. I use Zod as a schema validator. I created a schema that looks like this:

```typescript
const example = z.object({
  status: z.enum(["foo", "bar"]),
});
```

The form looks like this:

```typescript
const form = useForm({
  initialValues: {
    status: '',
  },
  validate: zodResolver(example),
});
```

I cant set the status to an empty string or null because then the type is inferred to be of that type and the `onSubmit((values) => ...)` function becomes useless because I would expect the type of the Zod schema there, not the type of `initialValues`.

I know i can set the type of `useForm` with `useForm<z.infer<typeof example>>(...)`, however then i cant set the value to an empty string or null because the schema restricts me from doing so. 

The only solution i have found is to let it infer the type and then `example.parse(values)` it again inside of the onSubmit function. This seems suboptimal because the hook should already know the type since I provided the zodResolver with my schema and the onSubmit function does the validation. In my opinion the type of `values` in `onSubmit` should be of the result type of the schema provided to zodResolver. 

Another solution could be to introduce a second generic to `useForm` that determines the type of values inside `onSubmit` but not the type of `initialValues`.

**Comment by yverby** [comment, marked as answer]
It is also possible to specify the type for a specific field, but it is more likely a workaround

```ts
const example = z.object({
  status: z.enum(["foo", "bar"]),
});

type Example = z.infer<typeof example>;

const form = useForm({
  initialValues: { status: '' as Example['status'] },
  validate: zodResolver(example),
});
```

**Reply by stj-be** [reply]
Okay thank you didnt think of that. It is a workaround though, should I convert this to an issue to let the devs see this and maybe find a fix?

**Reply by edwandr** [reply]
This is also a problem I have, it would be great to update useForm typing to handle that well

**Comment by uncomplexity** [comment]
also having this problem right now.

i think the input schema and output schema should be different.
